src/sentry/replays/usecases/replay_counts.py

```diff
@@ -56,6 +58,19 @@ def _get_replay_id_mappings(
         # if we want to validate list of replay_ids existence
         return {v: [v] for v in value}
 
+    # The client may or may not have narrowed the request by project-id. We have a
+    # defined set of issue-ids and can efficiently look-up their project-ids if the
+    # client did not provide them. This optimization saves a significant amount of
+    # time and memory when querying ClickHouse.
+    #
+    # NOTE: Possible to skip this. If the client did not set project_id = -1 then
+    # we could trust the client narrowed the project-ids correctly. However, this
+    # safety check is inexpensive and bad-actors (or malfunctioning clients) could
+    # provide every project_id manually.
+    if select_column == "issue.id":
+        qs = Group.objects.select_related("project").filter(id__in=value)
```

Comment:
is there a potential security issue here as it doesn't seem like we validate that the projects returned from here belong to the organization of the user?

That is, could you pass in an arbitrary group_id from another org and get results here? Worth writing a test for this case I think. This looks like a fantastic optimization though.

Reply:
Great catch.  Yeah this would have allowed users to read from other projects.  I'll add coverage for the case.